### DISCUSSION TITLE
python-socketio

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#210] [Q&A]
Hi, @wu-clan 

I've been testing WebSocket communication recently, so I'm studying the integration of python-socketio with our current project.
I was wondering if you have some time to help test it together? Thanks.

**Comment by downdawn** [comment]
The maintenance of the fastapi_socketio project is not very active, so I'm considering using the native python-socketio.

**Comment by wu-clan** [comment]
What is the application scenario you want to achieve? Chatting? Real-time push? Internet of Things?

**Reply by downdawn** [reply]
Chatting and Real-time push

**Reply by wu-clan** [reply]
My thoughts:

I might add websocket code to the repository instead of socketio

socketio code might just be posted as `gits`

They may be simple message pushes and not include chat scenarios (I have no experience with this)

**Reply by downdawn** [reply]
Using websocket directly will be more troublesome than socketio.
I have used flask to write chat, and I just need to adapt it to fastapi.
